kotlin-native/runtime/src/main/cpp/ObjCExportCoroutines.mm

```diff
@@ -69,4 +103,15 @@
   }
 }
 
+extern "C" void Kotlin_ObjCExport_resumeUnitContinuation(KRef continuation, id error) {
+  ObjHolder holder;
+
+  if (error != nullptr) {
+    KRef exception = Kotlin_ObjCExport_NSErrorAsException(error, holder.slot());
+    Kotlin_ObjCExport_resumeContinuationFailure(continuation, exception);
+  } else {
+    Kotlin_ObjCExport_resumeUnitContinuationSuccess(continuation);
+  }
+}
```

Comment:
This file contains a lot of tricky low-level unsafe details, coupled with the memory manager. Some of the possible mistakes can result in hard-to-debug crashes. So it is critically important to keep it as simple as possible.
Any idea how to avoid copy-pasting all the functions?

Reply:
We could get rid of a separate `Kotlin_ObjCExport_resumeUnitContinuation` and use `Kotlin_ObjCExport_resumeContinuation` everywhere by changing `ObjCExportCodeGenerator.generateUnitContinuationToRetainedCompletionConverter` and passing `Unit` singleton value as a parameter to `callFromBridge` if it's possible.

Reply:
Also `Kotlin_ObjCExport_createContinuationArgument` and its unit version are almost exactly the same, only the last line is different.

Reply:
Nope, the first suggestion isn't going to work because it will make the completion block exposed to Obj-C have two arguments again.

Reply:
> We could get rid of the separate `Kotlin_ObjCExport_resumeUnitContinuation` and use `Kotlin_ObjCExport_resumeContinuation` everywhere by changing `ObjCExportCodeGenerator.generateUnitContinuationToRetainedCompletionConverter` to pass `Unit` singleton value as a parameter to `callFromBridge` if it's possible.

This should work.

> Also `Kotlin_ObjCExport_createContinuationArgument` and its unit version are almost exactly the same, only the last line is different.

Yep. I guess it should be possible to implement the latter through the former, if we wrap one-argument completion to two-argument completion.

Reply:
Sounds good, although I could use some guidance on how to refer to `Unit` value to pass to `callFromBridge` as an argument.

Reply:
Try `kotlinReferenceToLocalObjC(codegen.theUnitInstanceRef.llvm)`.
As a side effect, `Unit` will be converted twice -- to Obj-C here, and then back in `Kotlin_ObjCExport_resumeContinuation`. To avoid this, you can move the conversion of `result` from C++ code to the generated code.

Reply:
Thanks, your suggestion plus a conditional (to only emit `Unit` if error is null) did the job, the first part of getting rid of `Kotlin_ObjCExport_resumeUnitContinuation` is thus complete.

Now I need to look into moving the conversion from C++ to generated code.

Reply:
Addressed the remaining TODO items.